src/interface/src/styleguide/file-upload-field/file-upload-field.component.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  Component,
+  EventEmitter,
+  HostBinding,
+  Input,
+  Output,
+} from '@angular/core';
+import { MatButtonModule } from '@angular/material/button';
+import { NgIf } from '@angular/common';
+import { MatIconModule } from '@angular/material/icon';
+import { ButtonComponent } from '../button/button.component';
+import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
+
+export type UploadStatus = 'default' | 'running' | 'uploaded' | 'failed';
+export type MimeType = {
+  mime: string;
+  suffix: string;
+};
+
+@Component({
+  selector: 'sg-file-upload',
+  standalone: true,
+  imports: [
+    MatButtonModule,
+    MatIconModule,
+    ButtonComponent,
+    MatProgressSpinnerModule,
+    NgIf,
+  ],
+  templateUrl: './file-upload-field.component.html',
+  styleUrl: './file-upload-field.component.scss',
+})
+export class FileUploadFieldComponent {
+  @Input() fieldLabel?: string;
+  @Input() placeholder = 'Choose a shape file to upload';
+  @Input() disabled = false;
+  @Input() required = true;
+  @Input() uploadStatus: UploadStatus = 'default';
+  @Input() supportedTypes: MimeType[] = [
+    { mime: 'application/zip', suffix: 'zip' },
+  ];
+  @Output() fileEvent = new EventEmitter<any>();
+
+  file: File | null = null;
+
+  onFileUploaded(event: Event) {
+    this.uploadStatus = 'running';
+    const target = event.target as HTMLInputElement;
+    const files = target.files as FileList;
+    if (target.files) {
+      this.file = files.item(0);
+      this.uploadStatus = 'uploaded';
```

Comment:
I'm really unsure about including these automatic uploadStatus settings. I think the first might be reasonable, but I think the parent component might have its own criteria about "failed" uploads, for example, and it should retain the ability to set those.

Reply:
I think this is good, but probably we want to include an uploadFailed emitter, for example if no `target.files`

Reply:
Sure. I was wondering if that should be a separate event, or if it might be better for the consumer to just receive fileEvent for success, failures, or other situations, but I'm open either way.